python/tank/util/storage_roots.py

```diff
@@ -366,22 +365,53 @@ def get_local_storages(self, sg_connection):
 
         # create the SG connection and query
         log.debug("Querying SG local storages...")
-        sg_storages = sg_connection.find("LocalStorage", [], local_storage_fields)
-        log.debug("Query returned %s storages." % (len(sg_storages)))
+
+        # try to query storage root fields for specific project
+        project_storages = sg_connection.find_one(
+            "Project",
+            [["id", "is", project_id]],
+            [
+                "sg_windows_storage_root_path",
+                "sg_mac_storage_root_path",
+                "sg_linux_storage_root_path",
+            ],
+        )
+
+        # fall back to default behaviour if we're not inside a project
+        # or if no project-specific storage has been set
+        if not project_storages or (
+            not project_storages.get("sg_windows_storage_root_path")
+            and not project_storages.get("sg_mac_storage_root_path")
+            and not project_storages.get("sg_linux_storage_root_path")
```

Comment:
Do we need to make sure that all 3 fields are filled really?
Like, let's say a studio uses only Linux. They might not want to fill the other fields. Still the info is present.

Same comment for the requested fields in the query.

Reply:
The second part of the if condition catches the case where none of the fields are filled (not if _some_ aren't filled). This happens when the user has set up the new project fields, but hasn't set up any specific storage root paths.